catalog/rest/catalog-rest-impl/pom.xml

```diff
@@ -159,6 +159,9 @@
                             !org.codice.ddf.platform.util,
                             *
                         </Import-Package>
+                        <Export-Package>
+                            org.codice.ddf.rest.impl.action.*
```

Comment:
Apologies- I think this is my fault since my initial comment was not clear and may have mislead you.

❓ / ✏️ I believe we discourage exporting concrete classes and extending them in other bundles. While this technically works - it tightly couples us to that concrete class (eg making it so that we cannot update the implementation or 
 we break other bundles if we do update it). 

There is currently a way to register an ActionProvider from[ a MetacardTransformer Service registration](https://github.com/codice/ddf/blob/master/catalog/rest/catalog-rest-impl/src/main/java/org/codice/ddf/rest/impl/action/ActionProviderRegistryProxy.java#L55) but currently the way to communicate which attribute it should check is non null [is only via extending a class](https://github.com/codice/ddf/blob/master/catalog/rest/catalog-rest-impl/src/main/java/org/codice/ddf/rest/impl/action/ActionProviderRegistryProxy.java#L112-L116) (which isn't a good pattern). I would propose that we add a better alternative into [this method](https://github.com/codice/ddf/blob/master/catalog/rest/catalog-rest-impl/src/main/java/org/codice/ddf/rest/impl/action/ActionProviderRegistryProxy.java#L111-L118) that checks for a service property to get the attribute name, [like this.](https://gist.github.com/rzwiefel/7cb82547209b2be1b0a596b5001df3f4).

This will then allow us to remove this export package, remove `KmlActionProvider` class entirely, and then [add a service property here](https://github.com/codice/ddf/pull/4504/files#diff-6cc024c6674018f1277a5fb3c78ac15dR57) specifying the attribute to check for non null (eg, `<entry key="can-handle-attribute" value="location"/>`

Reply:
we can move this to just a ✏️ for now